addons/safeStart/functions/fnc_missionTimeWarning.sqf

```diff
@@ -0,0 +1,74 @@
+/*
+ * Author: BadWolf
+ * Function used to start/stop mission length timer
+ * Sould only be called on the server
+ *
+ * Arguments:
+ * 0: Start mission length warnings <BOOL>
+ *
+ * Examples:
+ * [true] call potato_safeStart_fnc_missionTimeWarning;
+ * [false] call potato_safeStart_fnc_missionTimeWarning;
+ *
+ * Public: Yes
+ */
+
+#include "script_component.hpp"
+TRACE_1("Params",_this);
+
+if !(isServer) exitWith {};
+
+params [];
+
+missionNamespace setVariable [QGVAR(missionstartTime), CBA_missionTime, true];
+TRACE_1("Mission start time",QGVAR(missionstartTime));
+
+// 15 min Warning
+[
+    {
+        private _missionStartTime = missionNamespace getVariable QGVAR(missionstartTime);
+        private _missionLength = parseNumber getMissionConfigValue QEGVAR(missionTesting,missionTimeLength) * 60;
+        (CBA_missionTime >= (_missionStartTime + _missionLength - 900))
+    },
+    {
+        [
+            "potato_adminMsg",
+            [
+                "15 mins to mission end",
+                "Time Keeper"
+            ]
+        ] call CBA_fnc_globalEvent;
+        ["Time Keeper", "15 mins to mission end", 30] call BIS_fnc_curatorHint;
```

Comment:
I'm pretty sure `BIS_fnc_curatorHint` only has local effects, need to send via event to zeuses